File: backend/app/schemas/patient.py
from pydantic import BaseModel
from typing import List, Optional, Dict, Any
from uuid import UUID
from datetime import date, datetime
from .enums import ConsentType


class PatientProfileResponse(BaseModel):
    id: UUID
    user_id: UUID
    hospital_id: UUID
    assigned_doctor_id: Optional[UUID]
    date_of_birth: Optional[date]
    sex: Optional[str]
    height_cm: Optional[float]
    weight_kg: Optional[float]
    created_at: datetime

    class Config:
        from_attributes = True


class PatientUpdate(BaseModel):
    date_of_birth: Optional[date] = None
    sex: Optional[str] = None
    height_cm: Optional[float] = None
    weight_kg: Optional[float] = None


class MedicalHistoryResponse(BaseModel):
    id: UUID
    patient_id: UUID
    chronic_conditions: Optional[Dict[str, Any]]
    medications: Optional[Dict[str, Any]]
    allergies: Optional[Dict[str, Any]]
    surgeries: Optional[Dict[str, Any]]
    family_history: Optional[Dict[str, Any]]
    smoking_status: Optional[str]
    pregnancy_status: Optional[str]
    additional_notes: Optional[str]
    skipped: bool
    updated_at: datetime

    class Config:
        from_attributes = True


class MedicalHistoryUpdate(BaseModel):
    chronic_conditions: Optional[Dict[str, Any]] = None
    medications: Optional[Dict[str, Any]] = None
    allergies: Optional[Dict[str, Any]] = None
    surgeries: Optional[Dict[str, Any]] = None
    family_history: Optional[Dict[str, Any]] = None
    smoking_status: Optional[str] = None
    pregnancy_status: Optional[str] = None
    additional_notes: Optional[str] = None
    skipped: Optional[bool] = None


class ConsentResponse(BaseModel):
    id: UUID
    patient_id: UUID
    consent_type: ConsentType
    accepted: bool
    accepted_at: datetime

    class Config:
        from_attributes = True


class ConsentCreate(BaseModel):
    consent_type: ConsentType
    accepted: bool

File: frontend/src/types/patient.ts
import { ConsentType } from './enums';

export interface PatientProfile {
  id: string;
  user_id: string;
  hospital_id: string;
  assigned_doctor_id: string | null;
  date_of_birth: string | null;
  sex: string | null;
  height_cm: number | null;
  weight_kg: number | null;
  created_at: string;
}

export interface PatientUpdate {
  date_of_birth?: string | null;
  sex?: string | null;
  height_cm?: number | null;
  weight_kg?: number | null;
}

export interface MedicalHistory {
  id: string;
  patient_id: string;
  chronic_conditions: Record<string, unknown> | null;
  medications: Record<string, unknown> | null;
  allergies: Record<string, unknown> | null;
  surgeries: Record<string, unknown> | null;
  family_history: Record<string, unknown> | null;
  smoking_status: string | null;
  pregnancy_status: string | null;
  additional_notes: string | null;
  skipped: boolean;
  updated_at: string;
}

export interface MedicalHistoryUpdate {
  chronic_conditions?: Record<string, unknown> | null;
  medications?: Record<string, unknown> | null;
  allergies?: Record<string, unknown> | null;
  surgeries?: Record<string, unknown> | null;
  family_history?: Record<string, unknown> | null;
  smoking_status?: string | null;
  pregnancy_status?: string | null;
  additional_notes?: string | null;
  skipped?: boolean | null;
}

export interface Consent {
  id: string;
  patient_id: string;
  consent_type: ConsentType;
  accepted: boolean;
  accepted_at: string;
}

export interface ConsentCreate {
  consent_type: ConsentType;
  accepted: boolean;
}
